pkg/networkpath/traceroute/runner.go

```diff
@@ -249,3 +330,89 @@ func getHostname(ipAddr string) string {
 	}
 	return currHost
 }
+
+func getInterface(localAddr net.IP) (net.Interface, error) {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return net.Interface{}, fmt.Errorf("failed to get interfaces: %w", err)
+	}
+
+	for _, iface := range interfaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			fmt.Printf("failed to get iface addresses: %s\n", err.Error())
+			continue
+		}
+
+		for _, addr := range addrs {
+			ip, _, err := net.ParseCIDR(addr.String())
+			if err != nil {
+				fmt.Printf("failed to parse cidr for %s: %s", addr.String(), err.Error())
+				continue
+			}
+
+			if ip.Equal(localAddr) {
+				return iface, nil
+			}
+		}
+	}
+
+	return net.Interface{}, fmt.Errorf("failed to find matching interface for %q", localAddr.String())
+}
+
+func (r *Runner) resolveSubnetID(ctx context.Context, iface net.Interface) (string, error) {
```

Comment:
I think this code is from https://github.com/DataDog/datadog-agent/blob/main/pkg/network/tracer/gateway_lookup.go#L108? Any chance we can extract the subnet cache from there into a common place for use from here and the network tracer module?

Reply:
+1 for factoring into reusable code if there is no blocker to do so.